src/async_wrap.cc

```diff
@@ -604,11 +604,15 @@ void AsyncWrap::EmitDestroy(Environment* env, double async_id) {
   env->destroy_async_id_list()->push_back(async_id);
 }
 
+void AsyncWrap::AsyncReset(double execution_async_id, bool silent) {
+  AsyncReset(object(), execution_async_id, silent);
```

Comment:
I don't think this overload should be allowed. It explicitly reuses the resource, correct?

Reply:
@Drieger can you please answer this?

Reply:
I agree, this seems questionable – this kind of conflicts with the `AsyncWrap` design, and in particular, this conflicts with e.g. implementing some kind of `currentResource()` model à la https://github.com/nodejs/node/pull/21313 (which I think is the whole point of this PR?).

Reply:
This change is not needed at all for this PR. I'm pushing a commit to remove this.

Reply:
This should be fixed, I've removed these.

Reply:
seems like the removal got lost.